bittensor/_subtensor/subtensor_impl.py

```diff
@@ -729,6 +762,197 @@ def add_stake(
+                List of wallets to stake.
+            amounts (List[Union[Balance, float]]):
+                List of amounts to stake. If None, stake all to the first hotkey.
+            wait_for_inclusion (bool):
+                if set, waits for the extrinsic to enter a block before returning true, 
+                or returns false if the extrinsic fails to enter the block within the timeout.   
+            wait_for_finalization (bool):
+                if set, waits for the extrinsic to be finalized on the chain before returning true,
+                or returns false if the extrinsic fails to be finalized within the timeout.
+            prompt (bool):
+                If true, the call waits for confirmation from the user before proceeding.
+        Returns:
+            success (bool):
+                flag is true if extrinsic was finalized or included in the block.
+                flag is true if any wallet was staked.
+                If we did not wait for finalization / inclusion, the response is true.
+        """
+        if not isinstance(wallets, list):
+            raise TypeError("wallets must be a list of bittensor.wallet")
+        
+        if len(wallets) == 0:
+            return True
+
+        if amounts is not None and len(amounts) != len(wallets):
+            raise ValueError("amounts must be a list of the same length as wallets")
+
+        if amounts is not None and not all(isinstance(amount, (Balance, float)) for amount in amounts):
+            raise TypeError("amounts must be a [list of bittensor.Balance or float] or None")
+
+        if amounts is None:
+            amounts = [None] * len(wallets)
+        else:
+            # Convert to Balance
+            amounts = [bittensor.Balance.from_tao(amount) if isinstance(amount, float) else amount for amount in amounts ]
+
+            if sum(amount.tao for amount in amounts) == 0:
+                # Staking 0 tao
+                return True
+
+        wallet_0: 'bittensor.wallet' = wallets[0]
+        # Decrypt coldkey for all wallet(s) to use
+        wallet_0.coldkey
+
+        neurons = []
+        with bittensor.__console__.status(":satellite: Syncing with chain: [white]{}[/white] ...".format(self.network)):
+            old_balance = self.get_balance( wallet_0.coldkey.ss58_address )
+
+            for wallet in wallets:
+                neuron = self.neuron_for_pubkey( ss58_hotkey = wallet.hotkey.ss58_address )
+
+                if neuron.is_null:
+                    neurons.append( None )
+                    continue
+
+                neurons.append( neuron )
+
+        # Remove existential balance to keep key alive.
+        ## Keys must maintain a balance of at least 1000 rao to stay alive.
+        total_staking_rao = sum([amount.rao if amount is not None else 0 for amount in amounts])
+        if total_staking_rao == 0:
+            # Staking all to the first wallet.
+            if old_balance.rao > 1000:
```

Comment:
this was in the add_stake already. Something to do with a minimum balance for the chain to keep the coldkey account

Reply:
> and I am confused with the `# Staking all to the first wallet.` comment
This is from the if check, if the total staking amount from the args is 0, we assume all balance will be staked.
And this has to go to the first wallet because not every wallet can get all the stake.

Reply:
I think the user should specify where should the money go instead of having us to make assumption that the money is going to the first wallet (and how do you define the first wallet?)